core/che-core-api-dto/src/main/java/org/eclipse/che/dto/server/DtoFactory.java

```diff
@@ -386,8 +386,13 @@ public Gson getGson() {
     private <T> DtoProvider<T> getDtoProvider(Class<T> dtoInterface) {
         DtoProvider<?> dtoProvider = dtoInterface2Providers.get(dtoInterface);
         if (dtoProvider == null) {
-            throw new IllegalArgumentException("Unknown DTO type " + dtoInterface);
+            if (dtoInterface.isAnnotationPresent(DTO.class)) {
+                throw new IllegalArgumentException("Provider of implementation for DTO type " + dtoInterface + " isn't found");
+            } else {
+                throw new IllegalArgumentException("Is not a DTO type " + dtoInterface);
```

Comment:
Then there will be an IllegalArgumentException:  "Is not a DTO type class blah-blah", won't it?